static/docs/install/docker.md

```diff
@@ -0,0 +1,114 @@
+# Using Docker
+
+Currently we don't provide any official Docker image, however you can easily
+create it yourself.
+
+## Writing the Dockerfile
+
+We support several ways to install DVC, the two recommended ones are through
+`pip` and `conda`.
+
+You can base your image on the official Python one:
+
+```dockerfile
+FROM python
+
+# Remember to include the cloud providers that you plan to use,
+# for example: "dvc[s3]"
+RUN pip install dvc
+```
+
+Or use a Conda based image:
+
+```dockerfile
+FROM continuumio/miniconda3
+
+RUN conda install dvc -y --quiet
+```
+
+For a recommended reading about writing Dockerfiles for Python applications, you
+can take a look at https://pythonspeed.com/docker/
+
+## Using Docker for your development environment
+
+Since `dvc` is a command line application, we have several recommendations to
+enhance the experience:
+
+- Make sure your _locale_ is set to UTF-8
+- Install the Zsh shell and its respective autocompletion script.
+- Install `less` so you can pipe the output of `dvc pipeline show`.
+- Enable movements with `Ctrl + Arrow` by editing the _inputrc_ file.
+- Use `pip install dvc` at the end of the file for better caching.
+- Set your `WORKDIR` to the path where you are going to mount your project.
+
+If you are using Linux, remember to map your user to your container. This way,
+you make sure your files are still writable. For more information:
+https://docs.docker.com/engine/security/userns-remap/
+
+### Example Dockerfile
+
+The following is a **suggestion** for a Dockerfile to use during development:
+
+```dockerfile
+FROM python:3.7
+
+# Use UTF-8 as the default locale
+ENV LANG=C.UTF-8
+
+# Better movement with [Ctrl + →] and [Ctrl + ←]
+ENV INPUTRC=/etc/inputrc
+
+RUN set -ex \
+  && echo '"\e[1;5C": forward-word' >> /etc/inputrc \
+  && echo '"\e[1;5D": backward-word' >> /etc/inputrc
+
+# Set working directory.
+# Mount your project under the same path.
+WORKDIR /usr/src
+
+# Install useful programs
+RUN apt-get update -y && apt-get install -y \
+      zsh \
+      less \
+      git
+
+#  Set Zsh as the default shell
+SHELL ["/usr/bin/zsh", "-c"]
+
+# Enable autocompletion scripts for DVC
+# and rice the completion style with `zstyle`:
+# http://zsh.sourceforge.net/Doc/Release/Zsh-Modules.html#The-zsh_002fzutil-Module
+RUN wget \
```

Comment:
Should this happen after installing DVC? Seems like a more logical order.

Reply:
@jorgeorpinel , it is more logical but not cache efficient. :)
If you want to update DVC you will need to recompute only the last part

Reply:
Docker cache? Interesting. (Found some info about this in https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache.)

What do you mean by updating DVC though, because I don't think executing `pip update dvc` on the running container will have anything to do with the cache. Do you mean simply rebuilding the very same image? (Sorry, I'm familiar with Docket but haven't been using it much this year.)

Reply:
Actually, per the link I provided, it seems both ways would leverage the cache (if you mean simply rebuilding the same image when you know DVC has an update), because the instruction is exactly the same, so the cache is never invalidated:

> Starting with a parent image that is already in the cache, the next instruction is compared against all child images derived from that base image to see if one of them was built **using the exact same instruction**. If not, the cache is invalidated.

Unless I'm missing something. But if I'm right, I would insist to invert the order to make it more logical/easier for humans to read 🙂

Reply:
I see this is reversed in iterative/dvc/pull/2844 🙂